erpnext/public/js/controllers/taxes_and_totals.js

```diff
@@ -390,10 +390,21 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
 		this.frm.doc.grand_total = flt(tax_count
 			? this.frm.doc["taxes"][tax_count - 1].total + flt(this.frm.doc.rounding_adjustment)
 			: this.frm.doc.net_total);
-
+		var payment_status = true;
 		if(in_list(["Quotation", "Sales Order", "Delivery Note", "Sales Invoice"], this.frm.doc.doctype)) {
 			this.frm.doc.base_grand_total = (this.frm.doc.total_taxes_and_charges) ?
 				flt(this.frm.doc.grand_total * this.frm.doc.conversion_rate) : this.frm.doc.base_net_total;
+			if (this.frm.doc.is_pos && this.frm.doc.is_return) {
+				$.each(this.frm.doc['payments'] || [], function (index, data) {
+					if (payment_status) {
+						data.amount = me.frm.doc.base_grand_total;
```

Comment:
amount should be in invoice currency

Reply:
Does it mean to use `base_amount` field instead of `amount`?

Reply:
I mean it should be like `data.amount = me.frm.doc.grand_total;`

Reply:
Thanks for feedback Updated.